benchmark/util/tilecover.benchmark.cpp

```diff
@@ -22,11 +22,8 @@ static void TileCoverPitchedViewport(benchmark::State& state) {
     Transform transform;
     transform.resize({ 512, 512 });
     // slightly offset center so that tile order is better defined
-    transform.setLatLng({ 0.1, -0.1 });
-    transform.setZoom(8);
-    transform.setAngle(5.0);
-    transform.setPitch(40.0 * M_PI / 180.0);
-    
+    transform.jumpTo(CameraOptions().withCenter(LatLng { 0.1, -0.1 }).withZoom(8.0).withAngle(5.0).withPitch(40.0));
```

Comment:
In the past, we avoided designated initializer lists because certain compilers we targeted didn’t support them. Is that still the case, or could we use designated initializer lists instead of the builder pattern?

```cpp
transform.jumpTo(CameraOptions {
    .center = LatLng { 0.1, -0.1 },
    .zoom = 8.0,
    .angle = 5.0,
    .pitch = 40.0,
})
```

Both syntaxes provide more clarity than unlabeled constructor parameters, but I find designated initializer lists to be more elegant. We already use designated initializer lists liberally in Objective-C++ code in platform/{darwin,ios,macos}/, since Clang supports them as an extension to C++. For example:

https://github.com/mapbox/mapbox-gl-native/blob/0addd511d5fa503dc67d0773893c66c000e3ec98/platform/darwin/src/MGLGeometry.mm#L114-L119

Reply:
Though I'm quite fond of designated initializer lists too, this technique is a bit different as it links chained setters via a reference to the `CameraOptions` object, but the constructor still remains empty. In C++, support for [designated initializers](https://en.cppreference.com/w/cpp/language/aggregate_initialization) comes in C++20, so it'll take a while until all our supported platforms can get availability on this.

Reply:
For reference, [Clang has supported designated initializers in C++](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0329r0.pdf#page=2) for years as an extension to the C++ standard. GCC at some point introduced support for “trivial designated initializers”, which means the fields have to come in the same order as they’re declared, like in Swift.

/ref #26 #4722